modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala

```diff
@@ -197,16 +256,42 @@ object ObservationService {
           ${hour_angle_range_value.opt}
       """
 
-    def updates(SET: ObservationPropertiesInput): Option[NonEmptyList[AppliedFragment]] = {
-      val upExistence = sql"c_existence = $existence"
-      val upSubtitle = sql"c_subtitle = ${text_nonempty.opt}"
-      val upStatus = sql"c_status = $obs_status"
-      val upActive = sql"c_active_status = $obs_active_status"
+    def updates(SET: ObservationPropertiesInput): Either[String, Option[NonEmptyList[AppliedFragment]]] = {
+      val upExistence    = sql"c_existence = $existence"
+      val upSubtitle     = sql"c_subtitle = ${text_nonempty.opt}"
+      val upStatus       = sql"c_status = $obs_status"
+      val upActive       = sql"c_active_status = $obs_active_status"
+
+      val upCloud        = sql"c_cloud_extinction = $cloud_extinction"
+      val upImage        = sql"c_image_quality = $image_quality"
+      val upSky          = sql"c_sky_background = $sky_background"
+      val upWater        = sql"c_water_vapor = $water_vapor"
+
+      val upAirMassMin   = sql"c_air_mass_min = ${air_mass_range_value.opt}"
+      val upAirMassMax   = sql"c_air_mass_max = ${air_mass_range_value.opt}"
+      val upHourAngleMin = sql"c_hour_angle_min = ${hour_angle_range_value.opt}"
+      val upHourAngleMax = sql"c_hour_angle_max = ${hour_angle_range_value.opt}"
 
-      val upCloud = sql"c_cloud_extinction = $cloud_extinction"
-      val upImage = sql"c_image_quality = $image_quality"
-      val upSky = sql"c_sky_background = $sky_background"
-      val upWater = sql"c_water_vapor = $water_vapor"
+      def elevationRangeUps(e: ElevationRangeInput): Either[String, List[AppliedFragment]] = {
+        val airMass: List[AppliedFragment] =
+          List(
+            e.airMass.flatMap(_.minPosBigDecimal).map(v => upAirMassMin(v.some)),
+            e.airMass.flatMap(_.maxPosBigDecimal).map(v => upAirMassMax(v.some))
+          ).flattenOption
+
+        val hourAngle: List[AppliedFragment] =
+          List(
+            e.hourAngle.flatMap(_.minBigDecimal).map(v => upHourAngleMin(v.some)),
+            e.hourAngle.flatMap(_.maxBigDecimal).map(v => upHourAngleMax(v.some))
+          ).flattenOption
+
+        (airMass, hourAngle) match {
+          case (Nil, Nil) => List.empty[AppliedFragment].asRight
+          case (am,  Nil) => (upHourAngleMin(None) :: upHourAngleMax(None) :: am).asRight
```

Comment:
when you switch from one kind of elevation constraint to another, you have to wipe out the opposing constraint type